include/drivers/clock_control.h

```diff
@@ -103,9 +103,16 @@ struct clock_control_driver_api {
 
 /**
  * @brief Enable the clock of a sub-system controlled by the device
- * @param dev Pointer to the device structure for the clock controller driver
- * 	instance
- * @param sys A pointer to an opaque data representing the sub-system
+ *
+ * This function is blocking: on success, the clock is enabled and
+ * ready when it returns. Use clock_control_async_on() for
+ * non-blocking operation.
+ *
+ * @param dev Device structure whose driver controls the clock.
+ * @param sys Opaque data representing the clock.
```

Comment:
It's not clear what it is actually. It's probably something that is vendor dependent.

Reply:
I don't understand this comment. That's what "opaque data" means, no?

Reply:
I've updated the clock disable comment to use the same language